lib/spack/spack/test/cmd/list.py

```diff
@@ -134,3 +135,21 @@ def test_list_count(mock_packages):
     assert int(output.strip()) == len(
         [name for name in spack.repo.all_package_names() if "py-" in name]
     )
+
+
+# def test_list_repos(mock_packages, builder_test_repository):
+def test_list_repos():
+    with spack.repo.use_repositories(
+        os.path.join(spack.paths.repos_path, "builtin.mock"),
+        os.path.join(spack.paths.repos_path, "builder.test"),
+    ):
+        total_pkgs = len(list().strip().split())
+
+        mock_pkgs = len(list("-r", "builtin.mock").strip().split())
+        builder_pkgs = len(list("-r", "builder.test").strip().split())
+
+        assert builder_pkgs == 8
+        assert total_pkgs > mock_pkgs > builder_pkgs
```

Comment:
Are these tests stable enough? Isn't it `assert total_pkgs == mock_pkgs + builder_pkgs` that would test that these are subsets?